work_queue/src/work_queue_json.c

```diff
@@ -0,0 +1,341 @@
+    if(!json){
+        return NULL;
+    }
+
+    //validate json
+    if(validate_json(json, work_queue_properties)){
+        return NULL;
+    }
+
+    void *i = NULL;
+    void *j = NULL;
+    const char* key = jx_iterate_keys(json, &j);
+    struct jx* value = jx_iterate_values(json, &i);
+
+    while (key != NULL) {
+
+        if (!strcmp(key, "name")){
+            name = value->u.string_value;
+        }
+        else if (!strcmp(key, "port")){
+            port = value->u.integer_value;
+        }
+        else if (!strcmp(key, "priority")){
+            priority = value->u.integer_value;
+        }
+        else{
+            printf("Not necessary: %s\n", value->u.string_value);
+        }
+
+        key = jx_iterate_keys(json, &j);
+        value = jx_iterate_values(json, &i);
+
+    }    
+
+    if(port){
+
+        struct work_queue *workqueue = work_queue_create(port);
+
+        if(!workqueue){ 
+            return NULL; 
+        }        
+
+        if(name){
+            work_queue_specify_name(workqueue, name);
+        }
+        if(priority){
+            work_queue_specify_priority(workqueue, priority);
+        }
+    
+        return workqueue;
+
+    }
+
+    return NULL;
+
+}
+
+/** Submit a task to a queue.
+Once a task is submitted to a queue, it is not longer under the user's 
+control and should not be inspected until returned via @ref work_queue_wait.
+Once returned, it is safe to re-submit the same take object via 
+@ref work_queue_submit.
+@param q A work queue object.
+@param str A JSON description of a task. 
+
+{ "command_line" : "<i>string</i>" , "output_files" : <i>array of objects with one object per output file</i> -> 
+[ { "local" : "<i>string</i>" , "remote" : "<i>string</i>" , "flags" : "<i>integer</i>" } ] , "input _files" : 
+<i>array of objects with one object per input file</i> -> [ { "local" : "<i>string</i>" , "remote" : 
+"<i>string</i>" , "flags" : "<i>integer</i>" } ] , "tag" : "<i>string</i>" }
+
+@return An integer taskid assigned to the submitted task.
+*/
+int work_queue_json_submit(struct work_queue *q, const char* str){
+
+    struct work_queue_task * task;
+
+    task = create_task(str);
+
+    if(task){
+        return work_queue_submit(q, task);
+    }
+    else{
+        return -1;
+    }
+
+}
+
+/** Wait for a task to complete.
+@param q A work queue object.
+@param timeout The number of seconds to wait for a completed task before 
+returning. Use an integer time to set the timeout or the constant 
+@ref WORK_QUEUE_WAITFORTASK to block until a task has completed.
+@return A JSON description of the completed task or the
+ timeout was reached without a completed task, or there is completed child 
+process (call @ref process_wait to retrieve the status of the completed 
+child process).
+
+{ "command_line" : "<i>string</i>" , "tag" : "<i>string</i>" , "output" : "<i>string</i>" , "taskid" : 
+<i>integer</i> , "return_status" : <i>integer</i> }
```

Comment:
IIRC, a task has two returns, `exit_status` and `exit_code` (or something similar.)  One gives the disposition from a WQ perspective (complete, failed, removed), while the other gives the Unix exit status of the process, if it ran to completion.  Both shold be part of this result object.